lib/streamlit/elements/json.py

```diff
@@ -14,50 +14,65 @@
 
 import json
 import copy
-from typing import Any, cast, TYPE_CHECKING
+from typing import (
+    Any,
+    Hashable,
+    List,
+    MutableMapping,
+    Tuple,
+    Union,
+    cast,
+    TYPE_CHECKING,
+)
+
+from typing_extensions import TypeAlias
 
 from streamlit.proto.Json_pb2 import Json as JsonProto
 from streamlit.runtime.state import SessionStateProxy
+from streamlit.type_util import is_iterable
 from streamlit.user_info import UserInfoProxy
 
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
 
+Body: TypeAlias = Union[List[Any], Tuple[Any, ...], MutableMapping[Hashable, Any]]
 
-def _convert_sets_to_lists(body: Any) -> Any:
+
+def _convert_sets_to_lists(body: Body) -> Body:
     if isinstance(body, (list, tuple)):
         # We could technically iterate through the elements of a list/tuple and convert
         # any sets that we find to lists like we do below, but lists/tuples of sets
         # seem like a strange enough use-case that it's probably not worth the
         # additional complexity.
         return body
-
-    # Convert sets into lists, which render more nicely on the frontend
-    set_found = False
-    for key in body:
-        if isinstance(body[key], set):
-            if not set_found:
-                # When set is found to prevent mutation of input body, we need to shallow copy it once
-                # To avoid copying it multiple times we use set_found flag
-                body = copy.copy(body)
-                set_found = True
-            body[key] = list(body[key])
-    return body
+    else:
+        # Convert sets into lists, which render more nicely on the frontend
+        set_found = False
+        for key in body:
+            if isinstance(body[key], set):
+                if not set_found:
+                    # When set is found to prevent mutation of input body, we need
+                    # to shallow copy it once. To avoid copying it multiple times
+                    # we use set_found flag.
+                    body = copy.copy(body)
+                    set_found = True
+                body[key] = list(body[key])
+        return body
 
 
 class JsonMixin:
     def json(
         self,
-        body: Any,
+        body: object,
```

Comment:
From the outside perspective, `object` is equivalent with `Any` in the context of parameter types, as both allow passing in arbitrary objects. The difference is that the `object` type allows the code in the function declaration to be properly type checked, in the sense that it then has to gracefully support all objects.